superset/assets/javascripts/SqlLab/components/VisualizeModal.jsx

```diff
@@ -271,4 +270,19 @@ class VisualizeModal extends React.PureComponent {
 VisualizeModal.propTypes = propTypes;
 VisualizeModal.defaultProps = defaultProps;
 
-export default VisualizeModal;
+function mapStateToProps(state) {
+  return {
+    datasource: state.datasource,
+    errorMessage: state.errorMessage,
+  };
+}
+
+function mapDispatchToProps(dispatch) {
+  return {
+    actions: bindActionCreators(actions, dispatch),
+  };
+}
+
+export { VisualizeModal };
+export default connect(mapStateToProps, mapDispatchToProps)(VisualizeModal);
```

Comment:
that's a good article about presentational and container components. there are a lot of good points there, and i agree that it's an ideal architecture, separating out data fetching/updating logic from pure presentational components.

i think it would be worth doing a deep dive on the the component/redux architecture of sql lab, to see where we might optimize/organize which components will be containers, and which will be presentational. right now we have a bit of a mix happening. 

i would say connecting to redux from the visualizeModal component for now works, and allows us to better test that component by making that change. let's leave re-organizing for ideal container/presentation components for a follow up PR.